lib/thrift/include/thrift/transport/TSSLSocket.h

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright 2006 Facebook
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _THRIFT_TRANSPORT_TSSLSOCKET_H_
+#define _THRIFT_TRANSPORT_TSSLSOCKET_H_ 1
+
+// Put this first to avoid WIN32 build failure
+#include <thrift/transport/TSocket.h>
+
+// #include <openssl/ssl.h>
+#include <string>
+#include <thrift/concurrency/Mutex.h>
+
+#include <zephyr/net/tls_credentials.h>
+#include <zephyr/posix/sys/socket.h>
+
+namespace apache {
+namespace thrift {
+namespace transport {
+
+class AccessManager;
+#if 0
+class SSLContext;
+#endif
+
+enum SSLProtocol {
+  SSLTLS = 0,  // Supports SSLv2 and SSLv3 handshake but only negotiates at TLSv1_0 or later.
+               // SSLv2   = 1,  // HORRIBLY INSECURE!
```

Comment:
please don't comment-out SSLv2 - if it's unsupported, throw an exception in the implementation.

Reply:
I didn't modify this - it is commented out in the original source. In current implementation this enum type doesn't have any effect, as TLSv1_2 is hard coded into the open method.

Reply:
I see - I wouldn't necessarily hard-code it. I think throwing an exception might be better.

Reply:
Selection of TLS protocol is now implemented. It is wrapped in `SSLContext` as well.